src/R4Mvc.Tools/Services/PageGeneratorService.cs

```diff
@@ -290,7 +290,7 @@ private void AddParameterlessMethods(ClassBuilder genControllerClass, ITypeSymbo
                      */
                     .WithMethod(method.Key, "IActionResult", m => m
                         .WithModifiers(SyntaxKind.PublicKeyword, SyntaxKind.VirtualKeyword)
-                        .WithNonActionAttribute()
+                        .WithNonActionAndNonHandlerAttributes()
```

Comment:
And generate the `[NonHandler]` attribute here. I'm not sure if the `[NonAction]` actually does anything on razor pages. Could you confirm that please?

Reply:
Confirmed: only NonHandlerAttribute has an effect: https://github.com/aspnet/AspNetCore/blob/a159473c5729d4e334954d185c69ac144641685f/src/Mvc/Mvc.RazorPages/src/ApplicationModels/DefaultPageApplicationModelPartsProvider.cs#L158